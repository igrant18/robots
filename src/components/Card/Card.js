import React from 'react';

const Card = ({ name, email, id }) => {
    return (
        <div className='bg-light-green dib br3 pa3 ma2 grow bw2 shadow-5'>
            <img data-testid='id-img-element' alt='robot' src={`https://robohash.org/${id}?size=200x200`} />
            <div>
                <h2 data-testid='name-h2-element'>{name}</h2>
                <p data-testid='email-p-element'>{email}</p>
            </div>
        </div>
    )
}

export default Card;