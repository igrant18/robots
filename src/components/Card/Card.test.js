import '@testing-library/jest-dom';
// import React so you can use JSX (React.createElement) in your test
import React from 'react';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';

import Card from './Card';


describe('Testing Card Component', () => {

    const mockProp = {
        id: 1,
        name: 'John',
        email: 'Doe@gmail.com'
    }

    it('Renders Correctly with No Props', () => {
        const tree = renderer.create(<Card />).toJSON();
        expect(tree).toMatchSnapshot();
    });
    
    it('Displaying correct title', () => {

        render(<Card id={mockProp.id} name={mockProp.name} email={mockProp.email} />)
        expect(screen.getByTestId('name-h2-element')).toHaveTextContent(mockProp.name);
    });
});

