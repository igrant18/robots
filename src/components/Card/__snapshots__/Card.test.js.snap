// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Card Component Renders Correctly with No Props 1`] = `
<div
  className="bg-light-green dib br3 pa3 ma2 grow bw2 shadow-5"
>
  <img
    alt="robot"
    data-testid="id-img-element"
    src="https://robohash.org/undefined?size=200x200"
  />
  <div>
    <h2
      data-testid="name-h2-element"
    />
    <p
      data-testid="email-p-element"
    />
  </div>
</div>
`;
